src/Propel/Generator/Builder/Om/TableMapBuilder.php

```diff
@@ -464,40 +464,39 @@ protected function addNormalizedColumnNameMap(&$script): void
         $table = $this->getTable();
         $tableColumns = $table->getColumns();
 
+        $normalizedColumnNameMap = [];
+
         foreach ($tableColumns as $num => $column) {
             $normalizedName = strtoupper($column->getName());
 
             // ColumnName => COLUMN_NAME
-            $script .= '
-        \'' . $column->getPhpName() . '\' => \'' . $normalizedName . '\',';
+            $normalizedColumnNameMap[$column->getPhpName()] = $normalizedName;
 
             // TableName.ColumnName => COLUMN_NAME
-            $script .= '
-        \'' . $table->getPhpName() . '.' . $column->getPhpName() . '\' => \'' . $normalizedName . '\',';
+            $normalizedColumnNameMap[$table->getPhpName() . '.' . $column->getPhpName()] = $normalizedName;
```

Comment:
I updated the PR with a reduced version using array_walk but still using concatenation, IMHO better readable than regex and more stable for the admittedly very odd case of var_export changing it's output one day.

Reply:
That's pretty nice, too, I like it.

Reply:
Awesome, thanks!
I just fixed that one code sniffer issue. I also ran the sqlite unittests locally and they do not complain, however I couldn't wrap my past-midnight-head around writing a test for my fix. I want to, but there is only an internal method accessing the array and even if I could access that easily, it would see no difference as the duplicate keys are only visible in the source, not the interpreted code, the test would not fail for the code before the fix - it would be pointless.
The only idea I came up with is loading the generated file content as string and then check that a certain line appears exactly once. If that is not considered too dirty, I'm happy to provide that tomorrow.

Reply:
Right, that's a funny problem. The moment you can access the result, the problem is gone, one way or the other.

I like your idea of reading the php file into a string. There are the bookstore classes in `text/Fixtures/Bookstore` that you can use (they will be recreated every time the tests are started and the file `tests/Fixtures/fixtures_built` is not there).

Ultimately, the most important thing is that none of the column identifiers were removed, as this would really break things. But I assume that is tested somewhere already.

Another idea is to use the SchemaParser to create test data for the TableMapGenerator. I recently saw some tests that did that, now I know from where to copy. Otherwise, this is really no fun to figure out, so I'll go ahead and spoiler this: 
```php
    public function testNormalizedColumnMapHasUniqueKeys()
    {
        $databaseXml = '
<database>
    <table name="le_table">
        <column name="id" type="integer"/>
    </table>
</database>
';
        $reader = new SchemaReader();
        $schema = $reader->parseString($databaseXml);
        $table = $schema->getDatabase()->getTable('le_table');
        
        $tableMapBuilder = new class($table) extends TableMapBuilder{
            public function getNormalizedColumnNameMapDefinition(): string
            {
                $script = '';
                $this->addNormalizedColumnNameMap($script);
                return $script;
            }
        };
        $tableMapBuilder->setGeneratorConfig(new QuickGeneratorConfig());
        $normalizedColumnMapDefinition = $tableMapBuilder->getNormalizedColumnNameMapDefinition();
        
        $expectedColumnMapDefinition = <<< 'EOT'

    /**
     * Holds a list of column names and their normalized version.
     *
     * @var string[]
     */
    protected $normalizedColumnNameMap = [

        'Id' => 'ID',
        'LeTable.Id' => 'ID',
        'id' => 'ID',
        'leTable.id' => 'ID',
        'LeTableTableMap::COL_ID' => 'ID',
        'COL_ID' => 'ID',
        'le_table.id' => 'ID',
    ];

EOT;
        $this->assertEquals($expectedColumnMapDefinition, $normalizedColumnMapDefinition);
    }
```
Downside is, it is quite long, and will break anytime the output format is changed. In case you want to use it, it might need adjusting.

Stupid question: If that empty line before the first array element is still there, could I trouble you to get rid of it? Stuff like that makes me itchy. What kind of barbarians are we? If it remains, I'll live, of course.

Reply:
Thanks, that saved me a lot of time. I think we can meet in the middle. The "load the file" version does not cover any code which is also a code smell. But just comparing strings is open to a few flaws like you listed. I chose to extract the array elements from the generated code and check their number and the number of unique elements. That way the tests fail if something is disappearing (or appearing) and also if the elements should not be unique.

And no worries about that extra line, sorry I didn't catch that in the first place.

Reply:
And a great test case it is! Nice.

I think with the ms modifiers, you can simplify the regex to `'/= \[.*\]/ms'`. But that's nitty-gritty (or even wrong?).

Thank you for removing the extra line. My faith in humanity grew a notch.

If you like, you can squash your commits into one with git rebase and force-pushing. But one way or another, this is  perfect.

@dereuromark, do you mind approving this for running workflows?

Reply:
> I think with the ms modifiers, you can simplify the regex to '/= \[.*\]/ms'. But that's nitty-gritty (or even wrong?).

Well, true, but then I would have to trim the result as I split by "\n" further on... I wouldn't bet money on what is more performant...

also: commits squashed   --   good to go I hope, and glad I could finally contribute to my favorite ORM!

Reply:
Great!

And you are right, I overlooked the groups in the regex. The additional two lines are unique, so it would still work, and it comes down to a more readable regex vs more precise tests. And I like more precise tests. So again, thumbs are up.

Awesome, this was fun!